django_saml2_auth/views.py

```diff
@@ -152,10 +152,28 @@ def _create_new_user(username, email, firstname, lastname):
     user.save()
     return user
 
+def load_metadata(r, metadata_id):
+    metadata = SamlMetaData.objects.filter(pk=metadata_id).first()
+
+    if not metadata:
+        return HttpResponseRedirect(
+            get_reverse([denied, "denied", "django_saml2_auth:denied"])
+        )
+
+    return HttpResponse(
+        metadata.metadata_contents,
+        headers={"Content-Type": "text/xml; charset=utf-8"},
+    )
```

Comment:
there's something weird about exposing metadata files on the public internet just because this service itself needs access to them. why not keep them internal?

Reply:
swapped to using a `NamedTemporaryFile` to keep metadata files internal